src/util.rs

```diff
@@ -589,6 +589,22 @@ pub(crate) const fn min(a: NonZeroUsize, b: NonZeroUsize) -> NonZeroUsize {
     }
 }
 
+pub(crate) const fn assert_dst_is_not_zst<T>()
+where
+    T: crate::KnownLayout + ?Sized
+{
+    trait ConstAssert: crate::KnownLayout {
+        const DST_IS_NOT_ZST: bool = {
+            assert!(!Self::DST_IS_ZST);
+            !Self::DST_IS_ZST
+        };
```

Comment:
Couldn't we just do this based on the existing `KnownLayout::LAYOUT` const rather than introducing new machinery in `KnownLayout`?

Reply:
Done